### ISSUE TITLE
[94] Unified-IO: A Unified Model for Vision, Language, and Multi-Modal Tasks

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#124]
[paper](https://arxiv.org/pdf/2206.08916.pdf)

큰 scale 로 seq2seq 형태의 모델을 train 시킴.
GRIT benchmark 라고, AllenAI 쪽에서 공개한 multi-task benchmark 가 있는데, 하나의 모델로 이들 전부 task 를 풀 수 있는 work 는 처음이라고 주장함.
말 그대로 엄청나게 많은 task 들을 전부 풀어내는 형태.

# Related works
꽤 읽을만함.
[12-in-1](https://arxiv.org/abs/1912.02315) 같은 형태는 shared-backbone 을 활용함.
이는 task-wise specialized head 가 존재해야 함.

NLP 진영의 T5, GPT-3, Palm, Opt, gopher 등은 task-specific head 없이도 성공적으로 동작함.

Vision 쪽은 어찌 해야 할까.
text image input을 함께 주고, 처리시키는 방식은 꽤나 유명함. 
(cross attention 이든 simvlm 처럼 encoder 에서 짬뽕시키든 뭐든 간에 ㅇㅇ)
SimVLM, BLIP 등이 있는데, 얘네는 visual output 은 만들 수 없음 (segmentation 같은 거 안됨)

GPV-1 은 bbox 를 생성할 수 있도록 만들었고, (object detection)
GPV-2 는 GPV-1의 text decoder 를 사용해서 bbox input, bbox output 이 나올 수 있도록 만들었음 (region captioning 대응 가능)
VL-T5, OFA, pix2seq 와 같이 location token 을 활용할 수도 있긴 함.

Gato (A generalist agent) 는 atari game 에서의 button press 나, 로봇의 joint 움직임 등을 input 으로 받게끔 확장했고,
Flamingo 는 text, image, video 의 interleaved input 까지 지원함

다른 계열로 perceiver-io 나 uni-perceiver 도 있는데, 얘네도 generative task 를 풀지는 못함.

One-for-All (OFA) 가 concurrent work 로 있는데, unified-io 와 접근이 굉장히 비슷함.
다른 점은, 
1. OFA 는 CNN 을 썼고, 여기선 T5 를 사용
2. 여기서 더 큰 scale의 실험이 이루어짐.

UViM 도 concurrent work 인데, pretrained d-vae 대신 따로 second model 을 학습시키고, 여기서 훨씬 다양한 task 를 커버함.

# Model
![image](https://user-images.githubusercontent.com/16400591/175815252-fce707b5-e9c3-4ca7-91ba-78e9ed5c7c27.png)

4개의 모델 scale (T5)
1. XL - 24 layers 2.8B params
2. Large - 24 layers, 776M params
3. Base - 12 layers, 241M params
4. Small - 6 layers, 71M params

학습은 pretraining - multi-task finetuning 2-step 이고,
task-specific 하게 추가적으로 finetuning 을 수행하지는 않는다.
pretraining 은 다양한 objective 로 이루어진다.
1. text-only data (common-crawl) 에 대해서는 T5 방식의 학습
2. 이미지들에 대해서는 BEIT방식의 학습 (75% masking)
3. caption 데이터에 대해서는 "An image of" 를 prompt 로 붙이고, SimVLM 방식으로 학습시킴

multi-task training 에서는 80개의 public set 들을 죄다 때려박는 형태.
총 4가지 그룹이 될 수 있음
1. classical CV tasks
    1. image classification
    2. object detection
    3. instance segmentation
    4. depth/surface normal estimation
2. image synthesis tasks
    1. image in-painting
    2. image synthesis from caption/segmentation
3. V&L tasks
    1. VQA
    2. image captioning
    3. visual commonsense reasoning
4. NLP tasks
    1. GLUE
    2. Squad
    3. 등등

task 별 문제 정의 형태는 후술.
# Tasks
### Image Classification
ImageNet 2012, ImageNet21k, Places365, Sun397, iNaturalist, Caltech bird
![image](https://user-images.githubusercontent.com/16400591/175818914-86769b03-77ff-4663-97d5-0228570224b8.png)

### Object Categorization
Visual Genome, Open Images, and COCO
![image](https://user-images.githubusercontent.com/16400591/175818968-a1ae616f-ad19-4792-bd5c-aa48ae7b862a.png)

### Object Detection
Visual Genome, Open Images, and COCO
![image](https://user-images.githubusercontent.com/16400591/175818996-85e302d8-6a74-4d15-a182-016b1c7b4ca0.png)

### Object Localization
Visual Genome, Open Images, and COCO
![image](https://user-images.githubusercontent.com/16400591/175819035-c42ffffe-f771-4670-8f7b-0b9260c963a6.png)

### Keypoint Estimation
17 keypoints on a human body (e.g., eyes, nose, feet, etc.)
COCO human pose data
![image](https://user-images.githubusercontent.com/16400591/175819049-e413be8f-e6cc-4b48-9b13-e84b576848e3.png)

### Object Segmentation
Open Images, LVIS, COCO
![image](https://user-images.githubusercontent.com/16400591/175819141-f425422e-c8f7-48c5-b319-5c380620cbd2.png)

### Depth Estimation
NYU Depth Dataset V2
![image](https://user-images.githubusercontent.com/16400591/175819179-381ca845-54f3-4a58-a1c3-cfe59e6948e2.png)

### Surface Normal Estimation
FrameNet, BlendedMVS
![image](https://user-images.githubusercontent.com/16400591/175819196-904e0204-07e1-4a08-b711-e67834c8a040.png)

### Image Synthesis from Text.
COCO Caption,  Conceptual Captions 3M, Conceptual Captions 12M, RedCaps
![image](https://user-images.githubusercontent.com/16400591/175818824-ec69d22e-1592-484d-8e65-c9a04791da42.png)

### Image Inpainting
Open Images, Visual Genome, COCO
![image](https://user-images.githubusercontent.com/16400591/175818843-269c7e34-6d86-4150-9585-97cdddcabeac.png)

### Image Synthesis from Segmentation.
COCO, Open Images, LVIS
![image](https://user-images.githubusercontent.com/16400591/175818899-e003a047-aedc-4b8b-9a2c-e153f6b06b73.png)

### Visual Question Answering
VQA 2.0, Visual Genome, VizWiz-VQA, AOKVQA
![image](https://user-images.githubusercontent.com/16400591/175819226-abc9a558-d317-4dc8-9a9c-a9e206fdcb2d.png)

### Answer-Grounded Visual Question Answering
VizWiz-VQA
![image](https://user-images.githubusercontent.com/16400591/175819330-be7629a7-2f5a-462d-82f9-c1b0626e4532.png)

### Image Captioning
COCO Caption,  Conceptual Captions 3M, Conceptual Captions 12M, RedCaps
![image](https://user-images.githubusercontent.com/16400591/175819419-79ea402f-2ae8-4caf-8f62-4fe5c3489b2f.png)

### Region Captioning
Visual Genome
![image](https://user-images.githubusercontent.com/16400591/175819427-c01e67fa-bbf7-42f3-88c0-fbd96d0725b0.png)

### Referring Expression Comprehension
RefCOCO, RefCOCO+, RefCOCOg
![image](https://user-images.githubusercontent.com/16400591/175819459-9dbcf067-14b0-430e-8853-5d04b4f93aea.png)

### Relationship Detection
Visual Genome, Open Images 
![image](https://user-images.githubusercontent.com/16400591/175819467-90a49b22-5558-4f4e-930e-0b7c1ff6b0e7.png)

### Question Answering
SQuAD, SWAG, OpenBookQA,  Common Sense QA, FreebaseQA, Fever 1.0, SciQ, CosmoQA
![image](https://user-images.githubusercontent.com/16400591/175819238-5d47769f-5eff-4f03-b7d0-d641dc6fc840.png)

### Text Classification
GLUE, SuperGLUE, SNLI, SciTail
![image](https://user-images.githubusercontent.com/16400591/175819241-eb91e6dd-b7c3-48ee-a183-cf1ed9a4b811.png)

# Results
## GRIT
![image](https://user-images.githubusercontent.com/16400591/175817867-8362261c-6499-42d0-b824-1a20a8b8011f.png)

## Others
![image](https://user-images.githubusercontent.com/16400591/175817324-548e2c8d-f988-439b-a94a-f0343d33cb38.png)
VQA와 swig(Grounded Situation Recognition) 에서 성능이 좋은 편이다.
